query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -84,6 +84,13 @@ pub async fn get_many_records(
     // The should_batch has been adjusted to reflect that as a band-aid, but deeper investigation is necessary.
     match sql_info.max_bind_values {
         Some(chunk_size) if query_arguments.should_batch(chunk_size) => {
+            if query_arguments.cannot_batch() {
+                return Err(SqlError::BindValuesExceeded(
+                    "Your query cannot be split into multiple queries because of the order by aggregation or relevance"
```

Comment:
This error message doesn't apply to all non-batching cases. There will be more in the future as well, we need to restrict negation filters as well for example.